lib/agent.js

```diff
@@ -379,19 +362,22 @@ Agent.prototype._checkRequest = function(request) {
 // Handle an incoming message from the client
 Agent.prototype._handleMessage = function(request, callback) {
   try {
+    var plugin = this.actionToPlugin[request.a];
+
     var errMessage = this._checkRequest(request);
     if (errMessage) return callback(new ShareDBError(ERROR_CODE.ERR_MESSAGE_BADLY_FORMED, errMessage));
+    if (plugin) {
+      try {
+        plugin.checkRequest(request);
```

Comment:
Maybe this method should be optional? Or do we want to enforce that all plugins validate requests?